common/v2/features/SwapAssets/stateFactory.tsx

```diff
@@ -35,12 +35,17 @@ const SwapFlowFactory: TUseStateReducerFactory<SwapState> = ({ state, setState }
   const fetchSwapAssets = async () => {
     try {
       const assets = await DexService.instance.getTokenList();
+      const fromAsset = assets[0];
+      const sortedAssets = assets.sort((asset1: ISwapAsset, asset2: ISwapAsset) =>
+        (asset1.symbol as string).localeCompare(asset2.symbol)
+      );
+      const toAsset = assets[0];
       if (assets.length > 1) {
         setState((prevState: SwapState) => ({
           ...prevState,
-          assets,
-          fromAsset: assets[0],
-          toAsset: assets[1]
+          assets: sortedAssets,
+          fromAsset,
+          toAsset
         }));
       }
     } catch (e) {
```

Comment:
@FrederikBolding for future reference:
- When calling `setState` after an async call, make sure to cancel the request when the component un-mounts.
Otherwise we occasionally receive a `cannot setState on unmounted Component` error.
https://www.robinwieruch.de/react-warning-cant-call-setstate-on-an-unmounted-component
In the app we currently handle this case with `common/v2/vendor/react-use/usePromise.ts`, or any other https://github.com/streamich/react-use that is deemed useful.

- If we need to defend, we should do so as early as possible.
```
const assets = await DexService.instance.getTokenList();
if (assets.length < 1) return;
```

Suggestion:
We can wait to extract and test the sorting function. But in the meantime, either a comment or a more specific name - how do we sort - would be welcome.
```
   const sortedAssets = assets.sort((asset1: ISwapAsset, asset2: ISwapAsset) =>
        (asset1.symbol as string).localeCompare(asset2.symbol)
      );
```

Questions:
- Is it intentional we default to swapping the same asset?
```
const fromAsset = assets[0];
const toAsset = assets[0];
```
- And that we default to the first element in the returned list instead of the sorted list?
`const fromAsset = assets[0]` vs `const fromAsset = sortedAssets[0]`

Reply:
Fixed `cannot setState on unmounted Component` error, put a comment for sort function.
Regarding the questions, it was intentional because first returned asset was ETH, but I changed it, so we search for it now in case if dex.ag change the response in the future.
Thanks!